suspendmanager.lua

```diff
@@ -212,17 +243,40 @@ function shouldStaySuspended(job, accountblocking)
     end
 
     -- Internal reasons to be suspended, determined by suspendmanager
-    return shouldBeSuspended(job, accountblocking)
+    return self:shouldBeSuspended(job)
+end
+
+function SuspendManager:refresh()
+    self.jobsOnDesignation = {}
```

Comment:
note that some smooth/carve designations will still just be designations if they haven't been converted in to jobs yet. then you'd check the `smooth` field of the tile flags for smoothing and engraving and the occupancy.carve_track_* fields for carving tracks.

you can see examples of working with these values here: https://github.com/DFHack/scripts/blob/master/internal/quickfort/dig.lua (or https://github.com/DFHack/dfhack/blob/develop/plugins/dig-now.cpp)

Reply:
Added the check on designation/occupancy too, thanks